File: pset6/credit/credit.py
from cs50 import get_int

n = 0
while True:
    n = get_int("Number: ")
    if n > 0:
        break

tmp = n
dig = 0
while tmp != 0:
    tmp = tmp // 10
    dig += 1

card_alg = 0

for i in range(1, dig + 1):
    digs = n // 10**(i - 1)
    digs = digs % 10

    if i % 2 == 0:
        card_alg += (digs * 2) // 10 + (digs * 2) % 10
    else:
        card_alg += digs

flag = 0

first_one = (int)(n // 10**(dig - 1))
first_two = (int)(n // 10**(dig - 2))
if card_alg % 10 == 0:
    if dig == 15 and (first_two == 34 or first_two == 37):
        print("AMEX")
        flag = 1
    elif dig == 16 and (first_two == 51 or first_two == 52 or first_two == 53 or first_two == 54 or first_two == 55):
        print("MASTERCARD")
        flag = 2
    elif ((dig == 16 or dig == 13) and (first_one == 4)):
        print("VISA")
        flag = 3
    else:
        print("INVALID")
        flag = 4
else:
    print("INVALID")
    flag = 5

print(flag)

File: pset1/credit/credit.c
# include <stdio.h>
# include <cs50.h>
# include <math.h>

int main(void)
{
    long long  n;
    int dig = 0; //judge the digital number of n
    do
    {
        n = get_long_long("Number:");
    }
    while (n < 0);

    long long tmp;
    tmp = n;

    while (tmp != 0)
    {
        tmp = tmp / 10;
        dig = dig + 1;
    }
    //printf("digs = %d\n",dig);

    int card_alg = 0;

    for (int i = 1; i <= dig; i++)
    {
        long long digs;
        digs = n / pow(10, i - 1);
        digs = digs % 10;

        if (i % 2 == 0)
        {
            card_alg = card_alg + (digs * 2) / 10 + (digs * 2) % 10;
        }
        else
        {
            card_alg = card_alg + digs;
        }
    }
    //printf("%d\n",card_alg);

    int flag = 0;
    if (card_alg % 10 == 0)
    {
        //printf("n / pow(10 , dig-2) = %d\n",(int)(n / pow(10 , dig-2)));

        if ((dig == 15) & ((int)(n / pow(10, dig - 2)) == 34 || (int)(n / pow(10, dig - 2)) == 37))
        {
            printf("AMEX\n");
            flag = 1;
        }
        else if ((dig == 16) & ((int)(n / pow(10, dig - 2)) == 51 || (int)(n / pow(10, dig - 2)) == 52 || (int)(n / pow(10, dig - 2)) == 53|| (int)(n / pow(10, dig - 2)) == 54 || (int)(n / pow(10, dig - 2)) == 55))
        {
            printf("MASTERCARD\n");
            flag = 2;
        }
        else if ((dig == 16 || dig == 13) & ((int)(n / pow(10, dig - 1)) == 4))
        {
            printf("VISA\n");
            flag = 3;
        }
        else
        {
            printf("INVALID\n");
            flag = 4;
        }

    }
    else
    {
        printf("INVALID\n");
        flag = 5;
    }
    //printf("flag = %d\n",flag);

}